pennylane_lightning/core/src/utils/cuda_utils/cuGates_host.hpp

```diff
@@ -1437,6 +1437,54 @@ static constexpr auto getGeneratorIsingXY() -> std::vector<CFP_t> {
     };
 }
 
+/**
+ * @brief Create a matrix representation of the PSWAP gate data in row-major
+ * format.
+ *
+ * @tparam CFP_t Required precision of gate (`float` or `double`).
+ * @tparam U Required precision of parameter (`float` or `double`).
+ * @param angle Phase shift angle.
+ * @return std::vector<CFP_t> Return PSWAP gate data.
+ */
+template <class CFP_t, class U = double>
```

Comment:
Thank you @jzaia18 . I'm wondering if we can use the `custatevecApplyGeneralizedPermutationMatrix`API for the PSWAP gate implementation.

Reply:
Looking into it a bit, it seems like only the Generator versions of these gates use this API via `applyDevicePermutationGate_`. Even ordinary `applySWAP` uses `applyDeviceMatrixGate_`, any idea why that might be?

Reply:
Thanks @jzaia18 for looking into this. It's a legacy to use `applyDeviceMatrix` for all the gate operations. I'm wondering if `applyDevicePermutationGate_` would perform better than `applyDeviceMatrixGate_` for the gates like `SWAP` or `PSWAP`. If yes, I would recommend to use `applyDevicePermutationGate_` instead. CC: @maliasadi

Reply:
This will be addressed in a follow-up PR.